### DISCUSSION TITLE
Proposal: remove restriction that optional parameters must be trailing

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#8939] [Language Ideas]
Given that it is possible to used named parameters to specify arguments for parameters that appear after those for parameters with default values, the restriction that optional parameters must be trailing doesn't serve much of a purpose any longer.

**Comment by ygc369** [comment]
@gafter 
Please refer to #3293, similiar but a little different.

**Comment by jhv2017kam** [comment]
This may also resolve the issue of using optional parameters together with params (see also #1458)
